### PR TITLE
add the experiment phase as completed with error

### PR DESCRIPTION
**Pull request by ispeakc0de**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [ ] PR messages has document related information
-   [ ] Labelled this PR & related issue with `breaking-changes` tag
-   [ ] PR messages has breaking changes related information
-   [ ] Labelled this PR & related issue with `requires-upgrade` tag
-   [ ] PR messages has upgrade related information
-   [ ] Commit has unit tests
-   [ ] Commit has integration tests
-   [ ] E2E run Required for the changes

### CODE REVIEW

File: pkg/result/chaosresult.go [hunk 1]
```diff
@@ -308,13 +304,11 @@ func RecordAfterFailure(chaosDetails *types.ChaosDetails, resultDetails *types.R
 	}
 }
 
-func isProbeFailedErrorCode(errCode cerrors.ErrorType) bool {
-	switch errCode {
-	case cerrors.ErrorTypeK8sProbe, cerrors.ErrorTypePromProbe, cerrors.ErrorTypeCmdProbe, cerrors.ErrorTypeHttpProbe:
+func isProbeFailedErrorCode(reason string) bool {
```
Comment (avaakash) [inline pkg/result/chaosresult.go:307]: Why was this changed?

**Review by avaakash — APPROVED** [review]

**Review by neelanjan00 — APPROVED** [review]

File: pkg/result/chaosresult.go (hunk 1 above)
Comment (ispeakc0de) [reply pkg/result/chaosresult.go:307]: because if multiple probes fail (onchaos/continuous mode) then we are sending multiple error reasons in the form of [reasons, reason2] so it is unable to detect the probe error by error code.
